docs/nddata/mixins/ndarithmetic.rst

```diff
@@ -1,49 +1,406 @@
+  function. See the description for the mask-attribute for further details.
+
+wcs
+^^^
+
+The ``compare_wcs`` argument will determine what the result's ``wcs`` will be
+or if the operation should be forbidden. The possible values are identical to
+``mask`` and ``meta``:
+
+- If ``None`` the resulting ``wcs`` will be an empty ``None``.
+
+      >>> ndd1 = NDDataRef(1, wcs=0)
+      >>> ndd2 = NDDataRef(1, wcs=1)
+      >>> ndd1.add(ndd2, compare_wcs=None).wcs is None
+      True
+
+- If ``"first_found"`` or ``"ff"`` the resulting wcs will be the wcs of the
+  first operand or if that is None the meta of the second operand is
+  taken.
+
+      >>> ndd1 = NDDataRef(1, wcs=1)
+      >>> ndd2 = NDDataRef(1, wcs=0)
+      >>> ndd1.add(ndd2, compare_wcs='ff').wcs
+      1
+
+- If it's a ``callable`` it must take at least two arguments. Both ``wcs``
+  attributes will be passed to this function (even if one or both of them are
+  None) and the callable should return ``True`` if these wcs are identical
+  (enough) to allow the arithmetic operation or ``False`` if the arithmetic
+  operation should be aborted with a ``ValueError``. If ``True`` the ``wcs``
+  are identical and the first one is used for the result::
+
+      >>> def compare_wcs_scalar(wcs1, wcs2, allowed_deviation=0.1):
+      ...     if wcs1 is None and wcs2 is None:
+      ...         return True  # both have no WCS so they are identical
+      ...     if wcs1 is None or wcs2 is None:
+      ...         return False  # one has WCS, the other doesn't not possible
+      ...     else:
+      ...         return abs(wcs1 - wcs2) < allowed_deviation
+
+      >>> ndd1 = NDDataRef(1, wcs=1)
+      >>> ndd2 = NDDataRef(1, wcs=1)
+      >>> ndd1.subtract(ndd2, compare_wcs=compare_wcs_scalar).wcs
+      1
+
+  additional arguments can be passed in prefixing them with ``wcs_`` (this
+  prefix will be stripped away before passing it to the function)::
+
+      >>> ndd1 = NDDataRef(1, wcs=1)
+      >>> ndd2 = NDDataRef(1, wcs=2)
+      >>> ndd1.subtract(ndd2, compare_wcs=compare_wcs_scalar, wcs_allowed_deviation=2).wcs
+      1
+
+  If one is using `~astropy.wcs.WCS` objects a very handy function to use might
+  be::
+
+      >>> def wcs_compare(wcs1, wcs2, *args, **kwargs):
+      ...     return wcs1.wcs.compare(wcs2.wcs, *args, **kwargs)
+
+  see :meth:`astropy.wcs.Wcsprm.compare` for the arguments this comparison
+  allows.
+
+uncertainty
+^^^^^^^^^^^
+
+the ``propagate_uncertainties`` argument can be used to turn the propagation
+of uncertainties on or off.
+
+- If ``None`` the result will have no uncertainty::
+
+      >>> from astropy.nddata import StdDevUncertainty
+      >>> ndd1 = NDDataRef(1, uncertainty=StdDevUncertainty(0))
+      >>> ndd2 = NDDataRef(1, uncertainty=StdDevUncertainty(1))
+      >>> ndd1.add(ndd2, propagate_uncertainties=None).uncertainty is None
+      True
+
+- If ``False`` this is equivalent to ``"first_found"`` for ``meta``, ... and
```

Comment:
I would drop the part about being equivalent to `first_found` so that it reads "If ``False`` the result..."